gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandlerTest.java

```diff
@@ -0,0 +1,636 @@
+        inboundMessage.setEbXML(readLargeInboundMessageEbXmlFromFile());
+        var inboundMessageId = xPathService.getNodeValue(ebXmlDocument, "/Envelope/Header/MessageHeader/MessageData/MessageId");
+
+        when(patientAttachmentLogService.findAttachmentLog(inboundMessageId, CONVERSATION_ID))
+            .thenReturn(PatientAttachmentLog.builder()
+                .filename("test_frag_1.txt")
+                .mid("2").parentMid("1")
+                .patientMigrationReqId(1)
+                .build());
+
+        when(patientAttachmentLogService.findAttachmentLogs(CONVERSATION_ID))
+            .thenReturn(new ArrayList<>(
+                    Arrays.asList(
+                            PatientAttachmentLog.builder().filename("test_main.txt").mid("1")
+                                    .orderNum(0)
+                                    .parentMid("0")
+                                    .uploaded(false)
+                                    .largeAttachment(true)
+                                    .base64(true)
+                                    .compressed(false)
+                                    .contentType("text/plain")
+                                    .lengthNum(0)
+                                    .skeleton(false)
+                                    .patientMigrationReqId(1).build(),
+                            PatientAttachmentLog.builder().filename("test_frag_1.txt")
+                                    .mid("2")
+                                    .orderNum(0)
+                                    .parentMid("1")
+                                    .uploaded(true)
+                                    .largeAttachment(true)
+                                    .base64(true)
+                                    .compressed(false)
+                                    .contentType("text/plain")
+                                    .lengthNum(0)
+                                    .skeleton(false)
+                                    .patientMigrationReqId(1).build(),
+                            PatientAttachmentLog.builder().filename("test_frag_2.txt")
+                                    .mid("3")
+                                    .orderNum(1)
+                                    .parentMid("1")
+                                    .uploaded(true)
+                                    .patientMigrationReqId(1)
+                                    .largeAttachment(true)
+                                    .base64(true)
+                                    .compressed(false)
+                                    .contentType("text/plain")
+                                    .lengthNum(0)
+                                    .skeleton(false).build()
+                    )));
+
+        when(attachmentHandlerService.buildSingleFileStringFromPatientAttachmentLogs(any()))
+            .thenReturn("test-string");
+        copcMessageHandler.checkAndMergeFileParts(inboundMessage, CONVERSATION_ID);
+    }
```

Comment:
i think it is good practice to have an assertion for each unit test, perhaps even verifying that the mock services you've set up have been called would be better?

Reply:
I think Mockito fails tests which have any unused `when()` statements, so maybe there's no need to verify if a mock service has been called.